src/components/Modal/BaseModal.tsx

```diff
@@ -56,6 +56,7 @@ function BaseModal(
      */
     const hideModal = useCallback(
         (callHideCallback = true) => {
+            Modal.willAlertModalBecomeVisible(false);
```

Comment:
Same here. What does this change provide?

Reply:
Here I just moved `Modal.willAlertModalBecomeVisible(false)` from 2 `useEffect` hooks into `hideModal`.
This removes the race condition when double-clicking the `esc` button, as the `hideModal` is eventually called after each of those hooks, therefore `Modal.willAlertModalBecomeVisible(false)` will be called when the modal gets hidden. Without this change, the composer won't get focused when double-clicking `esc`.

I asked the original author here - there was no specific reason to put this call inside the hooks instead of the `hideModal`: https://github.com/Expensify/App/pull/27639#discussion_r1400271991